server/connectors/slack_connector/client.py

```
@@ -26,33 +26,50 @@ def __init__(self, access_token: str):
+                    if error == 'name_taken' and endpoint == 'conversations.create':
+                        pass
+                    else:
+                        logger.error(f"Slack API error on {endpoint}: {error}")
+                    raise ValueError(f"Slack API error: {error}")
+                
+                return result
+                
+            except requests.RequestException as e:
+                if attempt < max_retries and "429" in str(e):
+                    retry_after = min(2 * (attempt + 1), 30)
+                    logger.warning(f"Rate limited on {endpoint}, retrying in {retry_after}s (attempt {attempt + 1}/{max_retries})")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)